src/Swift.Runtime/tests/LibraryTests/SwiftSetTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Swift;
+using Swift.Runtime;
+using Xunit;
+
+namespace BindingsGeneration.Tests;
+
+public class SwiftSetTests : IClassFixture<SwiftSetTests.TestFixture>
+{
+    private readonly TestFixture _fixture;
+
+    public SwiftSetTests(TestFixture fixture)
+    {
+        _fixture = fixture;
+    }
+
+    public class TestFixture
+    {
+        static TestFixture()
+        {
+        }
+
+        private static void InitializeResources()
+        {
+        }
+    }
+
+    [Fact]
+    static void SmokeTest()
+    {
+        var metadata = TypeMetadata.GetTypeMetadataOrThrow<SwiftSet<SwiftIntMock>>();
+        // sizeof (Variant)
+        Assert.True(metadata.Size == 8);
+
+        var set = new SwiftSet<SwiftIntMock>();
```

Comment:
Good additional test would be to new up an array in C# and pass it to a swift method which validate that it works as an array correctly.

Reply:
Good point, added.

Reply:
Sorry - I don't see it. I was thinking something like:
```csharp
var set = new SwiftSet<SwiftIntMock>();
var sum = SumAllElementsInSwift(set);  // This would be a swift call which does foreach over the array and sums it up
Assert.Equal(sum, 0);
```

Reply:
Look at https://github.com/dotnet/runtimelab/pull/2968/files#diff-7c9e8d78468b134d5a41fcdc008666026d5335720c62f7959f1dd556a50ad8f8R361. It is in the functional tests under Structs, as it is a struct and there is infrastructure for compiling Swift code.

Extended with you case.